processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -257,9 +249,23 @@ func (c *WatchClient) Start() error {
 			return err
 		}
 		synced = append(synced, reg.HasSynced)
-		go c.nodeInformer.Run(c.stopCh)
+		waitForInformers = append(waitForInformers, c.nodeInformer)
+		go c.runInformer(c.nodeInformer)
 	}
 
+	reg, err = c.informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.handlePodAdd,
+		UpdateFunc: c.handlePodUpdate,
+		DeleteFunc: c.handlePodDelete,
+	})
+	if err != nil {
+		return err
+	}
+	synced = append(synced, reg.HasSynced)
```

Comment:
I think that eventually we should only block on the pod's `synced` result. The rest syncs are blocking pod's informer to start anyways, right?